Bottom: 542d0fcd16478312c482e5a629fdf0210d44077d
Top:    d6537d21636941387d5da0fb3a5e0e7bf90f4f51
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-26 12:01:35 +0800

vhost_net: conditionally start tx polling

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 9b51989..0390432 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -394,6 +394,7 @@ static void handle_tx(struct vhost_net *net)
 		goto out;
 
 	vhost_disable_notify(&net->dev, vq);
+	vhost_net_disable_vq(net, vq);
 
 	hdr_size = nvq->vhost_hlen;
 	zcopy = nvq->ubufs;
@@ -484,6 +485,8 @@ static void handle_tx(struct vhost_net *net)
 					% UIO_MAXIOV;
 			}
 			vhost_discard_vq_desc(vq, 1);
+			if (err == -EAGAIN)
+				vhost_net_enable_vq(net, vq);
 			break;
 		}
 		if (err != len)
